djcroco/tests/tests.py

```diff
@@ -1,54 +1,23 @@
-import time
-
+import mock
 from django.core.files.uploadedfile import SimpleUploadedFile
 from django.core.urlresolvers import reverse
-from django.utils import unittest
 from django.template import Context, Template
-from django.test.client import Client
+from django.utils import unittest
 
 from .models import Example, NullableExample
+from .utils import FakeCrocodocRequestMixin, TEST_DOC_DATA, TEST_DOC_NAME
 
 
-# simple 1-page pdf saying 'Hello, world!'
-TEST_DOC_DATA = (  # Multiline string, not tuple.
-    '%PDF-1.7\n\n1 0 obj  % entry point\n<<\n  /Type /Catalog\n  /Pages 2 0 '
-    'R\n>>\nendobj\n\n2 0 obj\n<<\n  /Type /Pages\n  /MediaBox [ 0 0 200 200 '
-    ']\n  /Count 1\n  /Kids [ 3 0 R ]\n>>\nendobj\n\n3 0 obj\n<<\n  /Type '
-    '/Page\n  /Parent 2 0 R\n  /Resources <<\n    /Font <<\n      /F1 4 0 R \n'
-    '    >>\n  >>\n  /Contents 5 0 R\n>>\nendobj\n\n4 0 obj\n<<\n  /Type '
-    '/Font\n  /Subtype /Type1\n  /BaseFont /Times-Roman\n>>\nendobj\n\n5 0 obj'
-    '  % page content\n<<\n  /Length 44\n>>\nstream\nBT\n70 50 TD\n/F1 12 '
-    'Tf\n(Hello, world!) Tj\nET\nendstream\nendobj\n\nxref\n0 6\n0000000000 '
-    '65535 f \n0000000010 00000 n \n0000000079 00000 n \n0000000173 00000 n '
-    '\n0000000301 00000 n \n0000000380 00000 n \ntrailer\n<<\n  /Size 6\n  '
-    '/Root 1 0 R\n>>\nstartxref\n492\n%%EOF\n'
-)
-TEST_DOC_NAME = 'test_doc_file.pdf'
-
-
-client = Client()
-
-
-def initial_setup():
-    """ Inits all here as we do not want doing it in *every* test """
-    # Create sample data
-    example = Example.objects.create(
-        name='Test item',
-        document=SimpleUploadedFile(TEST_DOC_NAME, TEST_DOC_DATA))
-
-    # Get data out of the model
-    instance = Example.objects.get(id=example.id)
-    return instance
-
-
-class CrocoTestCase(unittest.TestCase):
+class TestCrocoField(FakeCrocodocRequestMixin, unittest.TestCase):
     @classmethod
     def setUpClass(cls):
-        cls.instance = initial_setup()
-
-    def setUp(self):
-        # there is a race conditions somewhere so sleep between each test
-        time.sleep(1)
+        """Create sample data and store it on the class."""
+        super(TestCrocoField, cls).setUpClass()
+        example = Example.objects.create(
+            name='Test item',
+            document=SimpleUploadedFile(TEST_DOC_NAME, TEST_DOC_DATA),
+        )
+        cls.instance = Example.objects.get(id=example.id)
```

Comment:
`example` up there appears to be a `SimpleUploadedFile` for some reason, rather than an `Example`, and it doesn't have a lot of the required attributes.